packages/cli/bin/download-connector-list.js

```diff
@@ -0,0 +1,64 @@
+const https = require('https');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+
+const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
+
+const DEST = path.resolve('generators/datasource/connectors.json');
+const URL =
+  'https://raw.githubusercontent.com/strongloop/loopback-workspace/master/available-connectors.json';
+
+/**
+ * Function to dowload the list of available connectors from loopback-workspace
+ * so the list only has to be maintained in one place.
+ */
+async function download() {
+  var file = fs.createWriteStream(DEST);
+  var request = https
+    .get(URL, function(response) {
+      response.pipe(file);
+      file.on('finish', async function() {
+        file.close();
+        await transformConnectorJSON();
+      });
+    })
+    .on('error', function(err) {
+      fs.unlink(DEST);
+      return err;
+    });
+}
+
+/**
+ * This function transforms the array of Connector objects from
+ * loopback-workspace as follows:
+ *
+ * - Transforms the array into an object / map
+ * - Transforms display:password to type:password so it can be used by CLI directly
+ * - Transforms description to message so it can be used by CLI directly
+ */
+async function transformConnectorJSON() {
+  let data = await readFileAsync(DEST, 'utf-8');
```

Comment:
You are saving the downloaded JSON content to a file first, then read the file and overwrite its content in the second step. Have you considered passing the JSON content between `download` and `transformConnectorJSON` via in-memory value?

My recommendation is to use [request-promise-native](https://github.com/request/request-promise-native), it will take care of stream processing, error handling and flow control for you, and also add support for nice-to-have features like `process.env.PROXY`.

A mock-up implementation:

```ts
const data = await request(URL, {json: true});
const out = {};
data.forEach(...);
await writeFileAsync(DEST, JSON.stringify(out, null, 2));
```

Reply:
I had thought of using the request module but didn't want to introduce a dependency unnecessarily. Will add as a dev dependency and address this in a follow up PR.